arwen/host/arwen.go

```diff
@@ -42,7 +42,8 @@ func NewArwenVM(
 	cryptoHook vmcommon.CryptoHook,
 	vmType []byte,
 	blockGasLimit uint64,
-	gasSchedule map[string]map[string]uint64,
+	gasSchedule config.GasScheduleMap,
+	protocolReservedFunctions []string,
```

Comment:
can you make the input as the gasScheduleMap. a defined type, as we will have reservedFunctions, builtInFunctions and maybe others ?

Reply:
Fixed, `vmcommon.FunctionNames`.